### PR TITLE
Add pull_request_updater for Gitlab

### LINKED ISSUE #1603: [Feature Request] Rebase MR for GitLab
According to this https://github.com/dependabot/dependabot-script/issues/306 a `PullRequestUpdater` for GitLab should be writing to support this feature. Also the generic script in dependabot-script should be updated.

Useful links:
https://docs.gitlab.com/ee/api/merge_requests.html#rebase-a-merge-request
https://docs.gitlab.com/ee/api/commits.html#create-a-commit-with-multiple-files-and-actions

### PR DESCRIPTION
**Pull request by ahivert**
Thank you for creating dependabot! It's a nice project and so much usefull :star_struck: .

This PR will help to let dependabot rebase its branches to fix merge request conflicts.

Fixes #1603

We could then fix https://github.com/dependabot/dependabot-script/issues/306.

### CODE REVIEW

**Comment by gperdomor** [comment]
thx @ahivert ... I hope this will merged soon

**Comment by pkalemba** [comment]
@feelepxyz  can you look at this?

**Comment by Eugeny** [comment]
Would love to have this merged.

**Comment by ahivert** [comment]
For those who want to use it without have to wait, you can take inspiration here:
https://github.com/ahivert/kira-dependencies/tree/update-a-merge-request

**Comment by pkalemba** [comment]
@ahivert @greysteil  can i help you somehow to merge it?

**Comment by greysteil** [comment]
I don't work on Dependabot anymore, but this looks sensible to me (I haven't reviewed the code in detail). In the past, the thing that prevented us adding a `PullRequestUpdate` for GitLab is that it didn't support force pushes via the API. As it looks like that's now resolved, if the logic here mirros the logic in the GitHub one then it should be 💯.

Cc @feelepxyz @jurre

**Comment by pkalemba** [comment]
Cc @feelepxyz @jurre anybody?

**Comment by jurre** [comment]
Thanks for contributing this @ahivert, I've rerun the test suite as the failure looks like a flakey test, but I think overall the change looks good 👍

File: common/dependabot-common.gemspec [hunk 1]
```diff
@@ -37,6 +37,7 @@ Gem::Specification.new do |spec|
 
   spec.add_development_dependency "byebug", "~> 11.0"
   spec.add_development_dependency "gpgme", "~> 2.0"
+  spec.add_development_dependency "json", "~> 2.3"
```
Comment (feelepxyz) [inline common/dependabot-common.gemspec:40]: `json` should be included in the ruby std lib

```suggestion
```

**Pull request merged**